notebook/notebookapp.py

```diff
@@ -1100,6 +1109,13 @@ def start(self):
         
         This method takes no arguments so all configuration and initialization
         must be done prior to calling this method."""
+        try:
+            if os.geteuid() == 0 and not self.allow_root:
+                self.log.critical("Running as root is forbidden. Use --allow-root to bypass.")
+                self.exit(1)
+        except OSError as e:
+            pass
```

Comment:
No geteuid raise OSError on windows (according to the docs). So the try/catch is the dealing with windows case.